AzureStore.UI/app/src/components/Basket/Basket.tsx

```diff
@@ -44,40 +49,45 @@ const StyledTableRow = withStyles((theme: Theme) =>
 
 export default class Basket extends React.Component<Props> {
     state = {
-        orderItems : [{product:{id: 0, amount:0, name: ""}, quantity: 0}],
-        total: 0,
-        userEmail: ""
+      orderItems: new Array<OrderItem>(),
+      total: 0,
+      userEmail: ""
     };
 
     componentDidMount() {
       this.setState({
-        orderItems : this.props.selectedItems,
-        total: this.getTotal(this.props.selectedItems)
+        orderItems: this.props.selectedItems,
+        total: this.getTotal([...this.props.selectedItems])
```

Comment:
getTotal have access to this.props within its scope. No need to create new array and put it as parameter